metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistrationFencingChange.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.metadata;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+
+public enum BrokerRegistrationFencingChange {
+    REMOVED(-1, Optional.of(false)),
+    UNCHANGED(0, Optional.empty()),
+    ADDED(1, Optional.of(true));
```

Comment:
Maybe we should add JavaDoc comments to these enums. This is the documentation for that record:
> -1 if the broker has been unfenced, 0 if no change, 1 if the broker has been fenced.

I was interpreting REMOVED and ADDED as being removed from or added to the cluster. But it looks like we mean getting added to or removed from the fence state. The other option is to call this enum values FENCED, UNFENCED and NO_CHANGE. What do you think?

Reply:
ok, let's do FENCE, UNFENCE, NONE